### PR TITLE
Directed Cycle detection added in C++

### PR DESCRIPTION
**Pull request by YASH01009**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Description

(WoC)

### Checklist

- [Y] I've read the contribution guidelines.
- [Y] I've checked the issue list before deciding what to submit.
- [Y] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(#1468 Directed Graph Cycle Detection / WoC)

### CODE REVIEW

File: C-Plus-Plus/graphs/DirectedCycleDetection.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+#include <bits/stdc++.h>
+using namespace std;
+
+bool cycle;
+
+void dfs_visit(vector<vector<int>>& Graph, int src, vector<bool>& status, vector<bool>& visit) {
+	visit[src] = 1;
+	status[src] = 1;
+	for(int i = 0; i < Graph[src].size(); i++)
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/DirectedCycleDetection.cpp]: Add comments here explaining the Code

File: C-Plus-Plus/graphs/DirectedCycleDetection.cpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		    dfs_visit(Graph, Graph[src][i], status, visit);
+		else
+			if(status[Graph[src][i]] == 1)
+				cycle = 1;
+    status[src] = 0;
+}
+
+void dfs(vector<vector<int>>& Graph, int v, vector<bool>& status) {
+	vector<bool> visit(v);
+	for(int i = 0; i < v; i++)
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/DirectedCycleDetection.cpp]: Add comments here explaining the Code

**Review by HarshCasper — COMMENTED** [review]
Looks good to me! 

@iamrajiv Maybe you can take a look and merge it

**Comment by YASH01009** [comment]
@HarshCasper All the requested changes are done, now !!

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@YASH01009 Change the variable name `v` to `vertex` and `e` to `edge`.

**Comment by YASH01009** [comment]
@iamrajiv changes done !!

**Review by iamrajiv — APPROVED** [review]

**Pull request merged**